src/amuse/community/bhtree/tests/test_bhtree.py

```diff
@@ -1,929 +1,758 @@
+        10.0 | nbody_system.length
+    )
+    instance.commit_particles()
+    assert_equal(instance.get_radius(index), 10.0 | nbody_system.length)
+    assert_equal(instance.get_mass(index), 15.0 | nbody_system.mass)
+
+
+
+def test_test6(bhtree_kg):
+    instance = bhtree_kg
+    indices = instance.new_particle(
+        [15.0, 30.0] | units.kg,
+        [10.0, 20.0] | units.m, [20.0, 40.0] | units.m, [30.0, 50.0] | units.m,
+        [0.0, 0.01] | units.m/units.s, [0.0, 0.01] | units.m/units.s, [0.0, 0.01] | units.m/units.s,
+        [10.0, 20.0] | units.m
+    )
+    instance.commit_particles()
+
+    assert_equal(instance.get_mass(indices[0]), 15.0 | units.kg)
+
+    with pytest.raises(AmuseException) as excinfo:
+        instance.get_mass([4, 5])
+        assert "Error when calling 'get_mass' of a 'BHTree', errorcode is -1" in str(excinfo.value)
+
+
+@pytest.mark.parametrize(
+    "particle_fixture, raw_particle_data",
+    [(particle_inputs_kg, particle_inputs_kg)],
+    indirect=["particle_fixture"]
+)
+def test_test7(bhtree_kg, particle_fixture, raw_particle_data):
+    instance = bhtree_kg
+    instance.particles.add_particles(particle_fixture)
+    instance.commit_particles()
+
+    expected_count = raw_particle_data[0]
+    assert len(instance.particles) == expected_count
+
+    expected_mass = raw_particle_data[1]["mass"]
+    for idx in range(expected_count):
+        assert_equal(instance.get_mass(idx+1), expected_mass[idx])
+
+    expected_radius = raw_particle_data[1]["radius"]
+    for idx in range(expected_count):
+        assert_equal(instance.get_radius(idx+1), expected_radius[idx])
+
+    expected_position = raw_particle_data[1]["position"]
+    for idx in range(expected_count):
+        pos = instance.get_position(idx+1)
+        pos_e = expected_position[idx]
+        for x, x_e in zip(pos, pos_e):
+            assert_equal_with_reltol(x, x_e)
+
+    expected_velocity = raw_particle_data[1]["velocity"]
+    for idx in range(expected_count):
+        vel = instance.get_velocity(idx+1)
+        vel_e = expected_velocity[idx]
+        for x, x_e in zip(vel, vel_e):
+            assert_equal_with_reltol(x, x_e)
+
+
+@pytest.mark.parametrize(
+    "particle_fixture",
+    [particle_inputs_kg],
+    indirect=True
+)
+def test_test8(bhtree_kg, particle_fixture):
+    instance = bhtree_kg
+
+    instance.particles.add_particles(particle_fixture)
+    instance.commit_particles()
+
+    instance.particles.mass = [17.0, 33.0] | units.kg
+    assert_equal(instance.get_mass(1), 17.0 | units.kg)
+
+
+# NOTE: 0.125 is the default value for epsilon_squared;
+# it is put here to make things explicit.
+@pytest.mark.parametrize(
+    ("particle_fixture", "point", "epsilon2"),
+    [(particle_inputs_test9, 1.0, 0.00001), (particle_inputs_test10, 0.0, 0.125)],
+    indirect=["particle_fixture"]
+)
+def test_gravity_at_origin(make_bhtree, particle_fixture, point, epsilon2):
```

Comment:
Wait, isn't this actually testing that gravity at the center of gravity of a symmetric system is zero? For test9, we're not at the origin of the coordinate system but at (1, 0, 0).

Reply:
yes, you're right. I think I forgot to change the name after combining 2 tests into 1. I renamed to `test_zero_gravity`, which is what I think the test does for 2 different sets of parameters(?)